WikiDrips/WikiTableViewController.swift

```diff
@@ -49,12 +47,55 @@ class WikiTableViewController: UITableViewController, UISearchResultsUpdating {
             guard let strongSelf = self else { return }
             DispatchQueue.main.async(){
                 if newDocs.count > 0 {
+                    strongSelf.imageTasks.cancelAllOperations() // This doesn't seem to do anything yet
                     strongSelf.wikiDocs = [newDocs]
                     strongSelf.tableView.reloadData()
                 }
             }
         }
     }
+    
+    // MARK: - Image handling
+    
+    fileprivate let imageTasks = OperationQueue()
+    
+    fileprivate func generateImage(forItemAtIndex indexPath: IndexPath) {
+        guard safeIndexPath(indexPath: indexPath) else { return }
+        let wikiDoc = wikiDocs[indexPath.section][indexPath.row]
+        guard let initials = wikiDoc.imageInitials else {
+            return
+        }
+        
+        let generator = ImageGenerator(initials: initials)
+        imageTasks.addOperation(generator)
+        
+        generator.completionBlock = {
+            sleep(4)
+            OperationQueue.main.addOperation { [weak self] in
+//                guard let strongSelf = self else { return }
+                guard let image = generator.image else {
+                    print("Error creating image") // I'll change this to error logging latero
+                    return
+                }
+                // Following is still throwing "fatal error: Index out of range"
+                guard (self?.safeIndexPath(indexPath: indexPath) ?? false) && self?.wikiDocs[indexPath.section][indexPath.row].imageInitials == initials else { return } // Ignore because WikiDoc no longer matches
```

Comment:
When you get to the 'right' solution you won't need to check `imageInitials`. Also, these `guards` should all pass (meaning, when you do async correctly you won't be going in the `else` path). It's good to have them there but if you're hitting the `else` condition it's an indication that your async logic still needs work.